constrained-voucher.mkd

```diff
@@ -662,6 +669,38 @@ If the certificate that the Registrar uses is marked as a id-kp-cmcRA certificat
 
 In summary for typical Registrar use, where a single Registrar certificate is used, then the certificate MUST have EKU of: id-kp-cmcRA, id-kp-serverAuth, id-kp-clientAuth.
 
+## MASA Extensions {#brski-masa-extensions}
+
+This document supports efficient operation of Pledges in constrained networks, by letting a Pledge optionally omit the sending of its root CA certificate in the DTLS handshake with the Registrar.
+This optimization requires a Registrar to have an alternative method to obtain this root CA certificate, which it may need for validation and logging purposes. The alternative method defined here is 
+that the Registrar can request the relevant root CA certificate(s) directly from the MASA. Although the operator of a Registrar may have already obtained such CA certificate(s) via a vendor-specific method 
+(e.g. part of sales integration), a standardized method is still needed for cases where such vendor-specific methods do not exist or are not known to the operator.
+
+### Pledge CA Certificates Request
+A Registrar, after having established a TLS connection to the MASA, MAY request the full set of root CA certificates for all Pledges that are managed in scope of this MASA, by sending a Pledge CA 
+Certificates Request. This uses a HTTPS GET with the operation MASA URI path of "/.well-known/brski/cacerts".
+
+### Pledge CA Certificates Response
+A MASA server MUST support the HTTPS resource "/.well-known/brski/cacerts" for the GET method over the TLS connection where the client is a Registrar.
+
+If the request is processed successfully, the server response MUST have an HTTP 200 response code.  Any other response code indicates an error and the client MAY either retry the operation 
+or (eventually) abort the protocol.
+
+A successful response MUST be a certs-only CMC Simple PKI Response,
+as defined in {{RFC5272}}, containing the certificates described in the
+following paragraph.  The HTTP content-type of "application/pkcs7-mime" is used. The Simple PKI Response is sent
+with a Content-Transfer-Encoding of "base64" {{RFC2045}}.
+
+The MASA MUST include at least the root CA certificate(s) of the IDevID certificate chain(s) of any Pledge that the Registrar may potentially require.
+To determine what the Registrar may potentially require, without knowing yet the exact identity of the Pledge, the MASA may use one or more of the following context information:
```

Comment:
The Registrar probably already has the IDevID, which is how it got the MASAUrl.
Probably this request occurs before the requestvoucher?  
Could a third party do usefully use this?  Could a malicious party learn something from visiting it?

Reply:
The Registar indeed already has the IDevID of the Pledge, including things like serial-number and Authority Key Identifier. So it could make use of this information to signal to the MASA what root CA cert it wants to have.

That would be an extra barrier for a 3rd party to get certificate information.  Not sure if a malicious 3rd party would learn much from getting all the root CA certs though. 

One idea is the following: Registrar sends a POST request with a complete voucher-request object to the resource "/rootcacert" which returns in response the single X.509 root CA cert.  This is similar to how a Registrar sends the voucher-request object as an identifier to request the Pledge's audit log, in RFC 8995 5.8. The voucher-request object fully identifies the Pledge as well as the domain of the Registrar, which could be useful. And it allows code re-use I guess between MASA resource handlers.